google/cloud/storage/object_metadata.h

```diff
@@ -31,6 +32,15 @@ struct CustomerEncryption {
   std::string key_sha256;
 };
 
+/// Defines one of the source objects for a compose operation.
```

Comment:
@coryan does this design seem reasonable? I'm a little hesitant with the use of optional<>.

Reply:
LGTM. We are using `optional<>` in many other places.

Reply:
Sg, thanks @coryan! Addressed comments and fixed merge conflicts. Pending CI.